plenum/server/replica.py

```diff
@@ -380,12 +372,23 @@ def primaryName(self, value: Optional[str]) -> None:
         :param value: the value to set isPrimary to
         """
         self.primaryNames[self.viewNo] = value
+        self.compact_primary_names()
         if not value == self._primaryName:
             self._primaryName = value
             logger.debug("{} setting primaryName for view no {} to: {}".
                          format(self, self.viewNo, value))
             self._stateChanged()
 
+    def compact_primary_names(self):
+        min_allowed_view_no = self.viewNo - 1
```

Comment:
A simpler way is `self.primaryNames = {v: name for v, name in self.primaryNames.items() if v >= min_allowed_view_no}` but you would never have more than 2 items in this dict anyway